ducktape/command_line/main.py

```diff
@@ -109,7 +127,14 @@ def main():
 
     # Initializing the cluster is slow, so do so only if
     # tests are sure to be run
-    session_context.cluster = VagrantCluster()
+    try:
+        (cluster_mod_name, cluster_class_name) = args.cluster.rsplit('.', 1)
+        cluster_mod = importlib.import_module(cluster_mod_name)
+        cluster_class = getattr(cluster_mod, cluster_class_name)
+        session_context.cluster = cluster_class()
```

Comment:
Despite the recent type checking PEP, explicitly checking the type is discouraged, and shouldn't really be necessary here. They just need to have the right set of methods for us to invoke.